src/v/cloud_storage/remote_partition.h

```diff
@@ -178,7 +184,7 @@ class remote_partition
         _cvar.signal();
     }
 
-    using segment_map_t = std::map<model::offset, segment_state>;
+    using segment_map_t = absl::btree_map<model::offset, segment_state>;
```

Comment:
nit: this change may be a separate commit

Reply:
moved